### PR TITLE
Fix/make pytest works

### PR DESCRIPTION
**Pull request by benoitdm-oslandia**
* Add python requirements
* fix tests and logger usage
* add github CI actions

### CODE REVIEW

File: setup.cfg [hunk 1]
```diff
@@ -46,14 +46,24 @@ use_parentheses = True
...
 
-[tool:pytest]
-addopts =
-    --cov-config=setup.cfg
-    --cov=transapi
-    --ignore=tests/_wip/
-norecursedirs = .* build dev development dist docs CVS fixtures _darcs {arch} *.egg venv _wip
-python_files = test*.py
-testpaths = tests
+# CONFIGURATION MOVED TO pytest.ini: the pylint parser is unable to parse log_cli_format string because of the %
```
Comment (Djedouas) [inline setup.cfg:49]: is it the **pylint** parser? or the **pytest** parser?

File: requirements/development.txt [hunk 2]
```diff
@@ -0,0 +1,11 @@
+# Develoment dependencies
+# -----------------------
+
+black
+
+flake8>=3.8,<3.10
+flake8-builtins>=1.5,<1.6
+flake8-isort>=4.0,<4.1
+flake8-qgis>=1,<1.1
+
```
Comment (Djedouas) [inline requirements/development.txt]: flake8 -> pylint

File: setup.cfg (hunk 1 above)
Comment (benoitdm-oslandia) [reply setup.cfg:49]: Every config parser...
```
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/pip/_internal/cli/base_command.py", line 186, in _main
    status = self.run(options, args)
  File "/usr/lib/python3/dist-packages/pip/_internal/commands/install.py", line 433, in run
    lib_locations = get_lib_location_guesses(
  File "/usr/lib/python3/dist-packages/pip/_internal/commands/install.py", line 569, in get_lib_location_guesses
    scheme = distutils_scheme('', *args, **kwargs)
  File "/usr/lib/python3/dist-packages/pip/_internal/locations.py", line 109, in distutils_scheme
    d.parse_config_files()
  File "/usr/lib/python3.8/distutils/dist.py", line 413, in parse_config_files
    val = parser.get(section,opt)
  File "/usr/lib/python3.8/configparser.py", line 799, in get
    return self._interpolation.before_get(self, section, option, value,
  File "/usr/lib/python3.8/configparser.py", line 395, in before_get
    self._interpolate_some(parser, option, L, value, section, defaults, 1)
  File "/usr/lib/python3.8/configparser.py", line 442, in _interpolate_some
    raise InterpolationSyntaxError(
configparser.InterpolationSyntaxError: '%' must be followed by '%' or '(', found: '%d-%b-%y %H:%M:%S'
```

File: .github/workflows/pylint.yml [hunk 3]
```diff
@@ -30,4 +30,4 @@ jobs:
         pip install pylint
     - name: Analysing the code with pylint
       run: |
-        pylint $(git ls-files '*.py')
+        pylint --rcfile=./setup.cfg --fail-under=2.0 $(git ls-files 'TOMsPlugin/*.py') $(git ls-files 'tests/*.py')
```
Comment (Djedouas) [inline .github/workflows/pylint.yml]: In pre-commit config I set the --fail-under to 9.7 (current score is 9.71).

I think we can change it to --fail-under=8 here, and in pre-commit config file.

So we are sure to detect everything.

File: setup.cfg (hunk 1 above)
Comment (Djedouas) [reply setup.cfg:49]: I see ^^

File: .github/workflows/pylint.yml (hunk 3 above)
Comment (Djedouas) [reply .github/workflows/pylint.yml]: I just committed --fail-under=8 for pre-commit config file

**Review by Djedouas — APPROVED** [review]

**Pull request merged**